RabbitMQ.Stream.Client/Reliable/ReliableProducer.cs

```diff
@@ -116,19 +116,21 @@ protected override async Task GetNewReliable(bool boot)
                     await _producer.GetLastPublishingId());
             }
         }
-
-        catch (CreateProducerException ce)
-        {
-            LogEventSource.Log.LogError("ReliableProducer closed", ce);
-        }
         catch (Exception e)
         {
-            LogEventSource.Log.LogError("Error during initialization: ", e);
+            LogEventSource.Log.LogError("Error during producer initialization: ", e);
             SemaphoreSlim.Release();
+            if (await MaybeCloseDueToException(e))
+            {
+                throw;
+            }
+
             await TryToReconnect(_reliableProducerConfig.ReconnectStrategy);
         }
```

Comment:
During the ReliableProducer Creation `Task<ReliableProducer> CreateReliableProducer(...`  [the class raises an `Exception` if there is a protocol Error like `AccessDenied`](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/140/files#diff-163ebe022f4fea575575a9dff08283ef1aa3f65c1248e5cb6ec58ef53fefa5c0R98-R101) and the client is closed. something like:

```csharp
try
{
  var reliableProducer = await ReliableProducer.CreateReliableProducer(new ReliableProducerConfig()
     .....
} 
catch (Exception e)
{
    Console.WriteLine($"Error init  {e}");
}         
```


In other cases for example [Timeout connection](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/140#issuecomment-1162806469), the client [re-try to connect](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/140/files#diff-163ebe022f4fea575575a9dff08283ef1aa3f65c1248e5cb6ec58ef53fefa5c0R103) 


@ricsiLT suggested to avoid the reconnection during the init. If there is an error, just raise an Excpetion.

I am not totally sure about that. But we can discuss. 
WDYT @lukebakken ?

Reply:
In short, we should remove [`this TryToReconnect`](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/140/files#diff-b77a062150fa6a3bd56eaaaea8296e8bec8c09c63b7752370857a3b53955f6f3R128) 

Of course the same for the Consumer

Reply:
Per conversation with @lukebakken we are going to remove  [`this TryToReconnect`](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/140/files#diff-b77a062150fa6a3bd56eaaaea8296e8bec8c09c63b7752370857a3b53955f6f3R128)  and just throws an exception